hasktorch/src/Torch/Tensor.hs

```diff
@@ -320,7 +432,8 @@ instance {-# OVERLAPPING #-}TensorLike a => TensorLike [a] where
 
   asTensor v = asTensor' v defaultOpts
 
-  asValue t = unsafePerformIO $ do
+  asValue t' = unsafePerformIO $ do
+    let t = if isContiguous t' then t' else contiguous t'
```

Comment:
do we have a good understanding of which ops support non-contiguous tensors?

Reply:
https://discuss.pytorch.org/t/what-ops-make-a-tensor-non-contiguous/3676/6

Reply:
interesting, is it true though that every op in `Torch.Functional` can handle non-contiguous tensors as input?
I'm trying to figure out if we have to track "continuousness" at the type-level in `Torch.Typed`...